apollo-router/src/axum_factory/utils.rs

```diff
@@ -238,7 +238,9 @@ impl<B> MakeSpan<B> for PropagatingMakeSpan {
 
         // If there was no span from the request then it will default to the NOOP span.
         // Attaching the NOOP span has the effect of preventing further tracing.
-        if context.span().span_context().is_valid() {
+        if context.span().span_context().is_valid()
+            || context.span().span_context().trace_id() != opentelemetry::trace::TraceId::INVALID
```

Comment:
Wouldn't that second clause mean that we don't really need the first clause? What's the difference between checking that the context is valid and that the context trace ID isn't INVALID?

Reply:
what makes a valid span context when calling `span_context().is_valid()` is that the trace id AND span id should be valid. In case of the custom propagator, it exists because some legacy system only pass a trace id and not span id. Then we generate the span id if it's the case. That's why I check that the trace_id is always valid.